main/SelectBox/index.tsx

```diff
@@ -0,0 +1,194 @@
+import {
+  Animated,
+  Easing,
+  Platform,
+  StyleProp,
+  TextStyle,
+  TouchableOpacity,
+  View,
+  ViewStyle,
+} from 'react-native';
+import {DoobooTheme, light, useTheme} from '../theme';
+import React, {FC, ReactElement, useEffect, useRef, useState} from 'react';
+
+import {Icon} from '../Icon';
+import {Typography} from '../Typography';
+import styled from '@emotion/native';
+import {withTheme} from '@emotion/react';
+
+const Title = styled.View`
+  width: 200px;
+  height: 30px;
+  border-width: 1px;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+`;
+
+const Item = styled.View`
+  position: absolute;
+  width: 200px;
+  border-bottom-width: 1px;
+  border-left-width: 1px;
+  border-right-width: 1px;
+  justify-content: center;
+  align-items: center;
+`;
+
+type Styles = {
+  titleContainer?: StyleProp<ViewStyle>;
+  titleText?: StyleProp<TextStyle>;
+  rightElementContainer?: StyleProp<ViewStyle>;
+  itemContainer?: StyleProp<ViewStyle>;
+  itemText?: StyleProp<TextStyle>;
+};
+interface ItemCompProps {
+  value: string;
+  styles?: Styles;
+  itemHeight?: number;
+  translateYValue: number;
+  itemActiveOpacity: number;
+  onPress: (value: string) => void;
+}
+
+const ItemComp: FC<ItemCompProps> = ({
+  value,
+  styles,
+  itemHeight,
```

Comment:
Please refer to https://github.com/dooboolab/dooboo-ui/pull/67#discussion_r690609223. Never make style props as another props.

@yujong-lee I think it'd be awesome if we can have detailed code guidelines 😅 .

Reply:
@hyochan For sure. I will work on it little by little whenever I have time.

Reply:
> Great work!
> 
> After first glance of code, I got these thoughts in mind.
> 
> 1. I'm not sure, but have you think of `dropdown` as name? I want to hear your opinion on this!
> 2. I expect list should be collapsed on selection. I think at least option for this is needed.
> 3. Like [fix: [CheckboxStory]  responsive layout #61](https://github.com/dooboolab/dooboo-ui/pull/61), wrapping for mobile is needed.
> 4. Like [Remove responsibility for managing state from [ButtonGroup] #56](https://github.com/dooboolab/dooboo-ui/pull/56), I think responsible for managing selected item and handling click should be removed.
> 
> How to you think? 😄
> 
> ![스크린샷 2021-08-20 오후 10 46 35](https://user-images.githubusercontent.com/61503739/130243022-a6f1fcd7-2303-4b18-95ba-93e296201199.png)

I applied your suggestions. It's very good point. thanks.